test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -12,234 +12,235 @@
+						IPToPing = antreae2e.PodIPs{IPv6: &ip}
+					}
+					if err := e2eTestData.RunPingCommandFromTestPod(antreae2e.PodInfo{Name: sourcePod.podName, OS: osType, NodeName: sourcePod.nodeName, Namespace: namespace},
+						namespace, &IPToPing, containerName, pingCount, pingSize, false); err != nil {
+						return fmt.Errorf("ping '%s' -> '%s'(Interface: %s, IP Address: %s) failed: %v", sourcePod.podName, targetPod.podName, i, secondaryIPAddress, err)
 					}
+					logs.Infof("ping '%s' -> '%s'( Interface: %s, IP Address: %s): OK", sourcePod.podName, targetPod.podName, i, secondaryIPAddress)
 				}
 			}
 		}
 	}
 	return nil
 }
 
-// The Wrapper function createPodForSecondaryNetwork creates the Pod adding the annotation, arguments, commands, Node, container name,
-// resource requests and limits as arguments with the NewPodBuilder API
-func (data *TestData) createPodForSecondaryNetwork(ns string, nodeName string, podNum int, testPodName string, resNum int64) error {
-	computeResources := resource.NewQuantity(resNum, resource.DecimalSI)
-	return antreae2e.NewPodBuilder(podData[podNum].nameOfPods, ns, busyboxImage).OnNode(nodeName).WithContainerName(ctrName).WithCommand([]string{"sleep", "infinity"}).WithAnnotations(
-		map[string]string{
-			"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s", data.formAnnotationStringOfPod(podNum)),
-		}).WithLabels(
-		map[string]string{
-			"App": fmt.Sprintf("%s", testPodName),
-		}).WithResources(corev1.ResourceList{reqName: *computeResources}, corev1.ResourceList{reqName: *computeResources}).Create(data.e2eTestData)
-}
-
-func TestNativeSecondaryNetwork(t *testing.T) {
-	// once the setupTestWithSecondaryNetworkConfig is successful, we have all the prerequisites enabled and running.
-	_, err := testData.setupTestWithSecondaryNetworkConfig(t)
+func testSecondaryNetwork(t *testing.T, networkType string, pods []*testPodInfo) {
+	e2eTestData, err := antreae2e.SetupTest(t)
 	if err != nil {
-		t.Logf("Error when setupTestWithSecondaryNetworkConfig: %v", err)
+		t.Fatalf("Error when setting up test: %v", err)
 	}
+	defer antreae2e.TeardownTest(t, e2eTestData)
+
+	testData := &testData{e2eTestData: e2eTestData, networkType: networkType, pods: pods}
+
 	t.Run("testCreateTestPodOnNode", func(t *testing.T) {
-		testData.createPodOnNode(t, nameSpace, clusterInfo.controlPlaneNodeName)
+		testData.createPods(t, e2eTestData.GetTestNamespace())
 	})
 	t.Run("testpingBetweenInterfaces", func(t *testing.T) {
 		err := testData.pingBetweenInterfaces(t)
 		if err != nil {
-			t.Logf("Error when pinging between interfaces: %v", err)
+			t.Fatalf("Error when pinging between interfaces: %v", err)
 		}
 	})
 }
+
+func TestSriovNetwork(t *testing.T) {
+	// Create Pods on the control plane Node, assuming a single Node cluster for the SR-IOV
+	// test.
+	controlPlaneNodeName := antreae2e.NodeName(0)
```

Comment:
Is there a strong reason to specify that this is the "control-plane" Node, instead of just using `nodeName := antreae2e.NodeName(0)`?

Reply:
I was trying to keep the original code behavior. But is it clearer to call out the node is the control plane node?

Reply:
Usually you just need any Node, and the control-plane Node does not have any special meaning. In that case it's better to use `nodeName(0)`, or even not specify the Node unless you are specifically testing for inter-Node traffic.